third_party/CMakeLists.txt

```diff
@@ -19,6 +16,15 @@ if(${EFFCEE_ENABLE_TESTS})
 endif()
 
 if (IS_DIRECTORY ${EFFCEE_RE2_DIR})
+
+  if ("${MSVC}")
```

Comment:
We'd probably move this configuration into DXC itself. Instead of following effcee's original checkout location for re2, we can checkout re2 directly into DXC/external (and make sure re2 is `add_subdirecy` before effcee) like what we did for googletest. In that way, it's more clear and easier to manage all the external dependencies for DXC. This is also because effcee itself doesn't build with `/W4`, which is DXC actually.